setting-up-ocarina.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: docpost
+title: Setting up Ocarina and OAuth
+date_published: 
+date_modified:  2021-07-05 15:01:00 +0000
+author: BioWilko
+maintainer: BioWilko
+---
+
+### 0. Register for an account via Majora
+[Read the instructions for how to register](register)
+
+### 1. Install the Ocarina client
+
+[Ocarina](https://github.com/SamStudio8/ocarina/tree/master/ocarina) is a command line tool that is used to connect to Majora and perform actions with elevated privileges that are not possible on the website.
+It is not terribly difficult to use, but ideally you will have used a command line tool before.
+
+You can install the latest version with the Python package manager. You'll probably want to install it into a conda environment on the shared node.
+
+```
+conda create -n ocarina python=3.7
+conda activate ocarina
+pip install git+https://github.com/samstudio8/ocarina.git
+```
+
+Please remember to check the [ocarina changelog](https://github.com/SamStudio8/ocarina/blob/master/CHANGELOG.md) periodically and update your client as necessary.
+
+### 2. Register your Ocarina instance as an OAuth application
+
+Instructions to do this are located [here](oauth-app). Only the first section is necessary to set up ocarina but must be done before moving to the next step.
+
+### 3a. Set your Ocarina credentials with OAuth (Recommended)
+
+Create a JSON file in your home directory named `.ocarina`. Note the starting dot.
+
+```
+{"MAJORA_DOMAIN": "https://majora.covid19.climb.ac.uk/", "MAJORA_USER": "your-username", "MAJORA_TOKEN": "OAUTH", "CLIENT_ID": "your-client-id", "CLIENT_SECRET": "your-client-secret"}
+```
+
+Where:
+* `MAJORA_DOMAIN` points to either the domain of the real, or test Majora. Do not miss the `/` at the end of the URL.
+* `MAJORA_USER` is your username on the appropriate Majora instance
+* `MAJORA_TOKEN` must be set to `OAUTH`
+* `CLIENT_ID` is the ocarina Client ID registered with majora [accessible at this majora page](https://majora.covid19.climb.ac.uk/o/applications/)
+* `CLIENT_SECRET` is the ocarina Secret ID registered with majora [accessible at this majora page](https://majora.covid19.climb.ac.uk/o/applications/)
+
+OAuth has been the standard way to authenticate since September 2020 but legacy API keys are still supported, [instructions here.](getting-api-keys)
```

Comment:
remove this as there is a section for legacy keys

move the stuff from section 5 here > OAuth tokens expire after **10 hours after which you will have to repeat the process of generating a token by following step 4 again. See Section 5 for automating the token renewal process


instead add a note here to say that section 5 will explain how to update these